drivers/platform/maxim/maxim_rtc.c

```diff
@@ -0,0 +1,315 @@
+
+	mxc_rtc_regs_t *rtc_regs = MXC_RTC;
+	volatile uint32_t rtc_ctrl = rtc_regs->ctrl;
+	uint8_t n_int = 0;
+	/** Sub-second alarm flag clear */
+	rtc_regs->ctrl &= ~BIT(7);
+	/** Time-of-day alarm flag clear */
+	rtc_regs->ctrl &= ~BIT(6);
+	/** RTC (read) ready flag */
+	rtc_regs->ctrl &= ~BIT(5);
+
+	/** Shift right so the interrupt flags will be the first 3 bits */
+	rtc_ctrl >>= 5UL;
+	/** Clear the remaining bits */
+	rtc_ctrl &= 0x7UL;
+	while(rtc_ctrl) {
+		if((rtc_ctrl & 1) && (rtc_regs->ctrl & BIT(n_int))) {
+			cb->callback(cb->ctx, n_int, cb->config);
+		}
+		n_int++;
+		rtc_ctrl >>= 1;
+	}
+}
+
+/**
+ * @brief Initialize the RTC peripheral.
+ * @param device - The RTC descriptor.
+ * @param init_param - The structure that contains the RTC initialization.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_init(struct rtc_desc **device, struct rtc_init_param *init_param)
+{
+	int32_t ret = 0;
+	struct rtc_desc *dev;
+	sys_cfg_rtc_t sys_cfg;
+
+	if(!init_param)
+		return -EINVAL;
+
+	struct rtc_init_maxim *maxim_init_param = init_param->extra;
+	dev = calloc(1, sizeof(*dev));
+	if(!dev) {
+		ret = -ENOMEM;
+		goto error;
+	}
+
+	dev->id = init_param->id;
+	dev->freq = init_param->freq;
+	dev->load = init_param->load;
+
+	TMR_Enable(MXC_TMR0);
+
+	if(RTC_Init(MXC_RTC, dev->load, maxim_init_param->ms_load,
+		    &sys_cfg) != E_NO_ERROR) {
+		ret = -1;
+		goto error;
+	}
+
+	*device = dev;
+
+	return 0;
+
+error:
+	free(dev);
+
+	return ret;
+}
+
+/**
+ * @brief Free the resources allocated by rtc_init().
+ * @param dev - The RTC descriptor.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_remove(struct rtc_desc *dev)
+{
+	if(!dev)
+		return -EINVAL;
+	rtc_unregister_callback();
+	free(dev);
+	return 0;
+}
+
+/**
+ * @brief Start the real time clock.
+ * @param dev - The RTC descriptor.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_start(struct rtc_desc *dev)
+{
+	RTC_EnableRTCE(MXC_RTC);
+
+	/** Wait for synchronization */
+	if(RTC_CheckBusy())
+		return -EBUSY;
+
+	return 0;
+}
+
+/**
+ * @brief Stop the real time clock.
+ * @param dev - The RTC descriptor.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_stop(struct rtc_desc *dev)
+{
+	RTC_DisableRTCE(MXC_RTC);
+
+	return 0;
+}
+
+/**
+ * @brief Get the current count for the real time clock.
+ * @param dev - The RTC descriptor.
+ * @param tmr_cnt - Pointer where the read counter will be stored.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_get_cnt(struct rtc_desc *dev, uint32_t *tmr_cnt)
+{
+	if(RTC_CheckBusy())
+		return -EBUSY;
+
+	*tmr_cnt = RTC_GetSecond();
+
+	return 0;
+}
+
+/**
+ * @brief Set the current count for the real time clock.
+ * @param desc - The RTC descriptor.
+ * @param tmr_cnt - New value of the timer counter.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t rtc_set_cnt(struct rtc_desc *desc, uint32_t tmr_cnt)
```

Comment:
let's keep consistent, and use `dev` naming as in other functions from this file